packages/mds-db/index.ts

```diff
@@ -1221,16 +1221,37 @@ async function deleteGeography(geography_id: UUID) {
   return geography_id
 }
 
-async function publishGeography(geography_id: UUID) {
+async function publishGeography(params: { geography_id: UUID; effective_date: Timestamp }) {
   try {
     const client = await getWriteableClient()
-    const geography = await readSingleGeography(geography_id)
+    const geography = await readSingleGeography(params.geography_id)
     if (!geography) {
       throw new NotFoundError('cannot publish nonexistent geography')
     }
-    const sql = `UPDATE ${schema.TABLE.geographies} SET read_only = TRUE where geography_id='${geography_id}'`
-    await client.query(sql)
-    return geography_id
+    // set publish_date to now() if it isn't there already
+    // set effective_date to the provided param if it is earlier than what is already there
+    const vals = new SqlVals()
+    const conditions = []
+
+    if (!geography.read_only) {
+      conditions.push(`read_only = ${vals.add('t')}`)
+    }
+
+    if (!geography.publish_date) {
+      conditions.push(`publish_date = ${vals.add(now())}`)
```

Comment:
It seems like there's a possibility that `now()` used here for the geo and `now()` used in the `publish_date` for the policy could drift slightly, meaning that the `publish_date` timestamps wouldn't match. ??  I suggest you pass `publish_date` in as well as `effective_date`.